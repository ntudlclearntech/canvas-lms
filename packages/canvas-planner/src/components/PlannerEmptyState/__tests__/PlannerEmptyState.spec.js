/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that they will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import React from 'react';
import { shallow, mount } from 'enzyme';
import PlannerEmptyState from '../index';

function defaultProps(opts={}) {
  return {
    changeToDashboardCardView: ()=>{},
    onAddToDo: ()=>{},
    isCompletelyEmpty: true,
    ...opts
  };
}

it('renders desert when completely empty', () => {
  const wrapper = shallow(<PlannerEmptyState {...defaultProps()} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders balloons when not completely empty', () => {
  const wrapper = shallow(<PlannerEmptyState {...defaultProps({isCompletelyEmpty: false})} /> );
  expect(wrapper.find('.balloons').length).toEqual(1);
  expect(wrapper.find('.desert').length).toEqual(0);
});

it('does not changeToDashboardCardView on mount', () => {
  const mockDispatch = jest.fn();

  const changeToDashboardCardView = mockDispatch;

  mount(<PlannerEmptyState {...defaultProps({changeToDashboardCardView})} /> );
  expect(changeToDashboardCardView).not.toHaveBeenCalled();
});

it('calls changeToDashboardCardView on link click', () => {
  const mockDispatch = jest.fn();

  const changeToDashboardCardView = mockDispatch;

  const wrapper = mount(<PlannerEmptyState {...defaultProps({changeToDashboardCardView, isCompletelyEmpty:true})} /> );
  const button = wrapper.find('#PlannerEmptyState_CardView');

  button.simulate('click');
  expect(changeToDashboardCardView).toHaveBeenCalled();
});

it('does not call changeToDashboardCardView on false prop', () => {
  const wrapper = mount(<PlannerEmptyState {...defaultProps({isCompletelyEmpty:true})} /> );
  const button = wrapper.find('#PlannerEmptyState_CardView');

  button.simulate('click');
  expect(() => {
    button.simulate('click');
  }).not.toThrow();
});