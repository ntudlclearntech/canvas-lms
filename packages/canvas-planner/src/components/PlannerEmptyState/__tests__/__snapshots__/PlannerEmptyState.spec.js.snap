// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders desert when completely empty 1`] = `
<div
  className="PlannerEmptyState-styles__root planner-empty-state"
>
  <DesertSvg
    aria-hidden="true"
    className="PlannerEmptyState-styles__desert desert"
    height="220"
    viewBox="0 0 465 220"
    width="465"
    xmlns="http://www.w3.org/2000/svg"
  />
  <div
    className="PlannerEmptyState-styles__title"
  >
    <Heading
      border="none"
      color="inherit"
      level="h2"
    >
      No Due Dates Assigned
    </Heading>
  </div>
  <div
    className="PlannerEmptyState-styles__subtitlebox"
  >
    <div
      className="PlannerEmptyState-styles__subtitle"
    >
      Looks like there isn't anything here
    </div>
    <InlineList
      as="ul"
      delimiter="pipe"
      elementRef={[Function]}
      itemSpacing="none"
      margin="none"
      size="medium"
    >
      <InlineListItem
        delimiter="none"
        elementRef={[Function]}
        padding="none"
        size="medium"
        spacing="none"
      >
        <CondensedButton
          as="button"
          color="primary"
          cursor="pointer"
          elementRef={[Function]}
          id="PlannerEmptyState_CardView"
          margin="0"
          onClick={[Function]}
          size="medium"
          type="button"
        >
          Go to Card View Dashboard
        </CondensedButton>
      </InlineListItem>
      <InlineListItem
        delimiter="none"
        elementRef={[Function]}
        padding="none"
        size="medium"
        spacing="none"
      >
        <CondensedButton
          as="button"
          color="primary"
          cursor="pointer"
          elementRef={[Function]}
          id="PlannerEmptyState_AddToDo"
          margin="0"
          onClick={[Function]}
          size="medium"
          type="button"
        >
          Add To-Do
        </CondensedButton>
      </InlineListItem>
    </InlineList>
  </div>
</div>
`;
